### PR TITLE
add image capturing script for picamera

### PR DESCRIPTION
**Pull request by maxlou05**

### CODE REVIEW

File: picam.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+"""
+Taking images with a Raspberry pi camera module 3 on a Rpi on the drone.
+DO NOT ACTIVATE VENV WHEN USING THIS SCRIPT, IT IS MEANT TO BE USED WITH THE DEFAULT
+RASPBERRY PI LIBRARIES THAT ARE PRE-INSTALLED. Those libraries cannot be easily
+installed using pip nad requires additionaly steps, so we just use the provided
+library in global space.
```
Comment (Xierumeng) [inline picam.py]: Replace `Those libraries ...` with `See documentation for more detail.` and add this explanation to Confluence.

File: picam.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+"""
+
+import pathlib
+import time
+
+from picamera2 import Picamera2
```
Comment (Xierumeng) [inline picam.py]: `import picamera2`

File: picam.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+LOG_DIRECTORY_PATH = pathlib.Path("logs")
+SAVE_PREFIX = str(pathlib.Path(LOG_DIRECTORY_PATH, "image_" + str(int(time.time())) + "_"))
+DELAY = 1.0  # seconds
+
+
+def main() -> None:
+    """
+    Main loop for capturing and saving images.
+    """
+    picam2 = Picamera2()
```
Comment (Xierumeng) [inline picam.py]: `camera = picamera2.Picamera2()`

File: setup_project.ps1
Comment (Xierumeng) [inline setup_project.ps1:1]: Revert these changes.

File: setup_project.sh
Comment (Xierumeng) [inline setup_project.sh:1]: Revert these changes.

**Review by Xierumeng — COMMENTED** [review]
Reviewed.

File: picam.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
+"""
+Taking images with a Raspberry pi camera module 3 on a Rpi on the drone.
+DO NOT ACTIVATE VENV WHEN USING THIS SCRIPT, IT IS MEANT TO BE USED WITH THE DEFAULT
+RASPBERRY PI LIBRARIES THAT ARE PRE-INSTALLED. See documentation for more details.
+
+Or, when creating venv, use the --site-packages option
```
Comment (Xierumeng) [inline picam.py]: Update the documentation and then this can be removed.

File: rc.local.copyme [hunk 5]
```diff
@@ -8,3 +8,7 @@ chmod +x /home/warg/image-collection/run.sh  # Update to actual path
 # FFMPEG video (backup)
 chmod +x /home/warg/image-collection/video.sh
 /home/warg/image-collection/video.sh
+
+# Picamera2 (for use with RPi 5 and Picamera)
+cd /home/warg/image-collection
+python picam.py &
```
Comment (Xierumeng) [inline rc.local.copyme]: Move this into a bash script similar to `run.sh` and `video.sh` .

File: test_picam_settings.py [hunk 6]
```diff
@@ -0,0 +1,24 @@
+"""
+Utility script to test different picamera settings (for use on a Rpi5).
+See appendix C in https://datasheets.raspberrypi.com/camera/picamera2-manual.pdf
+for avilable configs.
```
Comment (Xierumeng) [inline test_picam_settings.py]: `For available settings, see appendix C: ...`

File: test_picam_settings.py [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+"""
+
+import picamera2
+
```
Comment (Xierumeng) [inline test_picam_settings.py:7]: 2 empty lines.

File: test_picam_settings.py [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+picam2 = picamera2.Picamera2()
```
Comment (Xierumeng) [inline test_picam_settings.py]: Add main guard.

File: test_picam_settings.py [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+
+picam2.configure(picam2.create_preview_configuration())
+# Example default settings for use with IR camera + IR filter. (only sees IR hotspot)
+# Don't change Brightness (artificially makes everything whiter) or Contrast (doesn't look too good)
+# picam2.set_controls({"ExposureTime": 250, "AnalogueGain": 64.0, "Brightness": 0.0, "Contrast": 1.0})
+print(picam2.controls)
+picam2.start(show_preview=True)
+
+while True:
+    command = input("enter a config and value (space separated): ")
```
Comment (Xierumeng) [inline test_picam_settings.py]: Does this block the preview?

Comment (Xierumeng) [inline test_picam_settings.py]: Capitalize: `Enter`

Add example input.

File: test_picam_settings.py [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+    config = command.split()[0]
+    value = float(command.split()[1])  # pylint: disable=invalid-name
+    if config == "ExposureTime":
+        value = int(value)  # pylint: disable=invalid-name
```
Comment (Xierumeng) [inline test_picam_settings.py]: Check length:
```py
input_tokens = command.split()
if len(input_tokens != 2):
    print("Invalid input")
    continue

config, raw_value = input_tokens
value = int(raw_value) if config == "ExposureTime" else float(raw_value)
```

File: test_picam_settings.py [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+    picam2.set_controls({command: value})
```
Comment (Xierumeng) [inline test_picam_settings.py]: Add empty line above this line.

Comment (Xierumeng) [inline test_picam_settings.py]: Add print of current:
```py
print("New controls set")
print(picam2.controls)
```

**Review by Xierumeng — COMMENTED** [review]
Reviewed.

File: test_picam_settings.py (hunk 9 above)
Comment (maxlou05) [reply test_picam_settings.py]: No, the preview opens first on line 16 and it continues working afterwards without issue

File: picam.sh
Comment (Xierumeng) [inline picam.sh:1]: Modify this to be in line with the other scripts (i.e. log and virtual environment).

File: test_picam_settings.py [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+"""
+
+import picamera2
+
+
+if __name__ == "__main__":
+    picam2 = picamera2.Picamera2()
+
+    picam2.configure(picam2.create_preview_configuration())
+    # Example default settings for use with IR camera + IR filter. (only sees IR hotspot)
```
Comment (Xierumeng) [inline test_picam_settings.py:14]: Is this comment for the line above?

File: test_picam_settings.py [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+    # Don't change Brightness (artificially makes everything whiter) or Contrast (doesn't look too good)
+    # picam2.set_controls({"ExposureTime": 250, "AnalogueGain": 64.0, "Brightness": 0.0, "Contrast": 1.0})
+    print(picam2.controls)
+    picam2.start(show_preview=True)
```
Comment (Xierumeng) [inline test_picam_settings.py:19]: Add comment that the preview is not blocked by the Python program.

**Review by Xierumeng — COMMENTED** [review]
Reviewed.

File: test_picam_settings.py (hunk 12 above)
Comment (maxlou05) [reply test_picam_settings.py:14]: It is for the line after (line 15)

File: picam.py [hunk 14]
```diff
@@ -0,0 +1,33 @@
...
+    name_counter = 0
+    while True:
+        picam2.capture_file(f"{SAVE_PREFIX}{name_counter}.png")
+        name_counter += 1
+
+        time.sleep(DELAY)
+
+
+if __name__ == "__main__":
+    LOG_DIRECTORY_PATH.mkdir(parents=True, exist_ok=True)
```
Comment (Xierumeng) [inline picam.py]: Move this into main function.

**Review by Xierumeng — COMMENTED** [review]
Reviewed.

**Review by Xierumeng — APPROVED** [review]
Approved, thanks!
